api/v2/log.go

```diff
@@ -9,9 +9,12 @@ import (
 // defaults to http.StatusInternalServerError. Fields is an optional set of
 // params provided in pairs, where the first of a pair is the key, and the second
 // is the value
-func (api *API) LogError(err error, message string, fields ...interface{}) func(c *gin.Context, code ...int) {
-	// create base entry
-	logger := api.l
+//
+// Passing in the initial gin.Context to LogError is used to extract the X-Request-ID associated
+// with the given request to make it easier to debug user-submitted erros.
+func (api *API) LogError(c *gin.Context, err error, message string, fields ...interface{}) func(c *gin.Context, code ...int) {
+	// create base entry with the associated request id
+	logger := api.l.Named(c.GetHeader("X-Request-ID"))
```

Comment:
`Named` is for components of a system I think. You probably want:
```suggestion
	logger := api.l.With("request-id", c.GetHeader("X-Request-ID"))
```